classes/queued_table.php

```diff
@@ -0,0 +1,129 @@
+ * @copyright  2023 Matthew Hilton <matthewhilton@catalyst-au.net>
+ * @license    http://www.gnu.org/copyleft/gpl.html GNU GPL v3 or later
+ */
+class queued_table extends table_sql {
+    /**
+     * Create table
+     * @param string $uniqueid
+     */
+    public function __construct(string $uniqueid) {
+        global $PAGE;
+
+        parent::__construct($uniqueid);
+
+        $columns = [
+            'id',
+            'timecreated',
+            'course',
+            'categoryfrom',
+            'categoryto',
+            'timestarted'
+        ];
+
+        $headers = [
+            get_string('queued:id', 'local_clonecategory'),
+            get_string('queued:timecreated', 'local_clonecategory'),
+            get_string('queued:course', 'local_clonecategory'),
+            get_string('queued:categoryfrom', 'local_clonecategory'),
+            get_string('queued:categoryto', 'local_clonecategory'),
+            get_string('queued:timestarted', 'local_clonecategory'),
+        ];
+
+        $this->define_columns($columns);
+        $this->define_headers($headers);
+        $this->baseurl = $PAGE->url;
+        $this->set_sql('id,customdata,timecreated,timestarted', '{task_adhoc}', "classname = :classname",
+            ['classname' => '\local_clonecategory\task\clone_course_task']);
+        $this->sortable(true, 'timecreated', SORT_DESC);
+    }
+
+    /**
+     * Course col
+     * @param object $row
+     */
+    public function col_course($row) {
+        $courseid = json_decode($row->customdata)->courseid;
+        $course = get_course($courseid);
+        return $course->fullname;
+    }
+
+    /**
+     * Category from col
+     * @param object $row
+     */
+    public function col_categoryfrom($row) {
+        return $this->get_cat_name(json_decode($row->customdata)->srcid);
+    }
+
+    /**
+     * Category to col
+     * @param object $row
+     */
+    public function col_categoryto($row) {
+        return $this->get_cat_name(json_decode($row->customdata)->destid);
+    }
+
+    /**
+     * Gets the name of category, or empty string if does not exist.
+     * @param int $catid id of category
+     * @return string
+     */
+    private function get_cat_name(int $catid): string {
+        $category = \core_course_category::get($catid);
+        return !empty($category) ? $category->name : '';
+    }
+
+    /**
+     * Time created col
+     * @param object $row
+     */
+    public function col_timecreated($row) {
+        // Don't show timecreated if timestarted set.
+        // Since the timecreated gets updated when the task begins.
+        // Might be confusing for users.
+        return empty($row->timestarted) ? userdate($row->timecreated) : get_string('taskstarted', 'local_clonecategory');
+    }
+
+    /**
+     * Time started col
+     * @param object $row
+     */
+    public function col_timestarted($row) {
+        return !empty($row->timestarted) ? userdate($row->timestarted) : '';
```

Comment:
instead of just a timestamp this might be better as a delta using format_time - but keep the timestamp as a title so so you have a tooltip of the timestamp